akka-projection-grpc/src/main/scala/akka/projection/grpc/internal/EventPusherConsumerServiceImpl.scala

```diff
@@ -116,11 +129,17 @@ private[akka] final class EventPusherConsumerServiceImpl(eventProducerDestinatio
                       transformedEventEnvelope.persistenceId,
                       if (transformedEventEnvelope.filtered) " filtered" else "")
 
+                  val writeEvent =
+                    if (transformedEventEnvelope.isEventDeserialized)
+                      transformedEventEnvelope.eventOption.getOrElse(FilteredPayload)
+                    else
+                      transformedEventEnvelope.serializedEvent.get
```

Comment:
the idea is that we can pass `SerializedEvent` to the EventWriter, and then the journal will understand that type and don't have to serialize.